libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -140,36 +142,45 @@ void IRGeneratorForStatements::initializeStateVar(VariableDeclaration const& _va
 	if (_varDecl.value())
 	{
 		_varDecl.value()->accept(*this);
-		string value = m_context.newYulVariable();
-		Type const& varType = *_varDecl.type();
-
-		m_code << "let " << value << " := " << expressionAsType(*_varDecl.value(), varType) << "\n";
-		m_code << IRStorageItem{m_context, _varDecl}.storeValue(value, varType);
+		writeToLValue(IRLValue{
+			*_varDecl.annotation().type,
+			IRLValue::Storage{
+				util::toCompactHexWithPrefix(m_context.storageLocationOfVariable(_varDecl).first),
+				m_context.storageLocationOfVariable(_varDecl).second
+			}
+		}, *_varDecl.value());
+		m_currentLValue.reset();
 	}
 }
 
 void IRGeneratorForStatements::endVisit(VariableDeclarationStatement const& _varDeclStatement)
 {
-	for (auto const& decl: _varDeclStatement.declarations())
-		if (decl)
-			m_context.addLocalVariable(*decl);
-
 	if (Expression const* expression = _varDeclStatement.initialValue())
 	{
-		solUnimplementedAssert(_varDeclStatement.declarations().size() == 1, "");
-
-		VariableDeclaration const& varDecl = *_varDeclStatement.declarations().front();
-		m_code <<
-			"let " <<
-			m_context.localVariableName(varDecl) <<
-			" := " <<
-			expressionAsType(*expression, *varDecl.type()) <<
-			"\n";
+		if (_varDeclStatement.declarations().size() > 1)
+		{
+			auto const* tupleType = dynamic_cast<TupleType const*>(expression->annotation().type);
+			solAssert(tupleType, "Expected expression of tuple type.");
+			solAssert(_varDeclStatement.declarations().size() == tupleType->components().size(), "Invalid number of tuple components.");
+			for (size_t i = 0; i < _varDeclStatement.declarations().size(); ++i)
```

Comment:
Why can't we do
```
for (auto [decl, value]: {_varDeclStatement.declarations(), tupleType->components()})
{
...
}
```
?
Maybe we should declare a helper `zip(CollectionA _a, CollectionB _b) { ... }`

Reply:
(although zip is not the right word)

Reply:
I'd suggest to think about that in general separately from this PR... I tried to define such a thing for different projects - it needs quite a bit of care...